### PR TITLE
Add Get|Set-ArubaSWPort

### PR DESCRIPTION
**Pull request by alagoutte**
Add cmdlets for Get and Configure Port (name, config, status, flow control, dsnoop trust...)

### CODE REVIEW

**Comment by alagoutte** [comment]
@CedricMoreau can you look ?

File: PowerArubaSW/Public/Port.ps1 [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+#
+
+function Get-ArubaSWPort {
+
+    <#
+        .SYNOPSIS
+        Get Port of the Switch
+
+        .DESCRIPTION
+        Get Portinformation
```
Comment (CedricMoreau) [inline PowerArubaSW/Public/Port.ps1]: Add a space ? (Between port and informtion)

File: PowerArubaSW/Public/Port.ps1 [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+    End {
+    }
+}
+
+function Get-ArubaSWPortStatistics {
+
+    <#
+        .SYNOPSIS
+        Get Port Sstatisticsof the Switch
```
Comment (CedricMoreau) [inline PowerArubaSW/Public/Port.ps1]: Add a space

File: PowerArubaSW/Public/Port.ps1 [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+        .DESCRIPTION
+        Get Portinformation
```
Comment (CedricMoreau) [inline PowerArubaSW/Public/Port.ps1]: Add a space

File: PowerArubaSW/Public/Port.ps1 [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('is_dsnoop_port_trusted') ) {
+            if ( $is_dsnoop_port_trusted ) {
+                $_port | add-member -name "is_dsnoop_port_trusted" -membertype NoteProperty -Value $true
+            } else {
+                $_port | add-member -name "is_dsnoop_port_trusted" -membertype NoteProperty -Value $false
+            }
+        }
```
Comment (CedricMoreau) [inline PowerArubaSW/Public/Port.ps1:198]: Add a blank line maybe ?

File: PowerArubaSW/Public/Port.ps1 [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        $response = invoke-ArubaSWWebRequest -method "PUT" -body $_port -url $url
+
+        $response | convertfrom-json
```
Comment (CedricMoreau) [inline PowerArubaSW/Public/Port.ps1:202]: Add a blank line too

File: Tests/integration/Port.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+. ../common.ps1
+#TODO: Add check if no ipaddress/login/password info...
+
+Describe  "Get Port" {
+    BeforeAll {
+        #Always set name to DEFAULT_PORT (no way to remove Port name/description...)
+        Set-ArubaSWPort -port_id 3 -name DEFAULT_PORT
+        #Always enable the port (disabled on WorkBench...)
+        Set-ArubaSWPort -port_id 3 -is_port_enable
+    }
```
Comment (CedricMoreau) [inline Tests/integration/Port.Tests.ps1:16]: Add a blank line ?

File: Tests/integration/Port.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+    It "Get Port Does not throw an error" {
+        {
+            Get-ArubaSWPort
+        } | Should Not Throw
+    }
+
+    It "Get ALL Port" {
+            $PORTS = Get-ArubaSWPort
```
Comment (CedricMoreau) [inline Tests/integration/Port.Tests.ps1]: Indent ? (not the same as below )

File: Tests/integration/Port.Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,181 @@
+#
+# Copyright 2018, Alexis La Goutte <alexis.lagoutte at gmail dot com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+. ../common.ps1
+#TODO: Add check if no ipaddress/login/password info...
+
+Describe  "Get Port" {
+    BeforeAll {
```
Comment (CedricMoreau) [inline Tests/integration/Port.Tests.ps1:11]: Add a blank line (or remove it at line 43)

File: Tests/integration/Port.Tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+        $PORT.bytes_tx | Should -Not -BeNullOrEmpty
+        $PORT.bytes_rx | Should -Not -BeNullOrEmpty
+        $PORT.throughput_tx_bps | Should -Not -BeNullOrEmpty
+        $PORT.throughput_rx_bps | Should -Not -BeNullOrEmpty
+        $PORT.error_tx | Should -Not -BeNullOrEmpty
+        $PORT.error_rx | Should -Not -BeNullOrEmpty
+        $PORT.drop_tx | Should -Not -BeNullOrEmpty
+        $PORT.port_speed_mbps | Should -Not -BeNullOrEmpty
+    }
+}
```
Comment (CedricMoreau) [inline Tests/integration/Port.Tests.ps1:183]: Add a blank line ?

File: Tests/integration/Port.Tests.ps1 [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+Describe  "Get Port Statistics" {
+
+    It "Get Port Does not throw an error" {
+        {
+            Get-ArubaSWPortStatistics
+        } | Should Not Throw
+    }
+
+    It "Get ALL Port Stastistics" {
+            $PORTS = Get-ArubaSWPortStatistics
```
Comment (CedricMoreau) [inline Tests/integration/Port.Tests.ps1]: Indent ?

**Pull request merged**